text/0118-search-api.md

```diff
@@ -460,14 +460,182 @@ Search queries using `_geoPoint` returns a `_geoDistance` field containing the d
+            "_formatted": {
+                "id": "2",
+                "title": "Pride and Prejudice"
+            }
+        },
+        {
+            "id": 456,
+            "title": "Le Petit Prince",
+            "_formatted": {
+                "id": "456",
+                "title": "Le Petit Prince",
+            }
+        }
+    ],
+    ...
+}
+```
+
+Which attributes are present in `_formatted`?
+
+The `_formatted` object will contain attributes coming from the original document, depending on the parameters the users set during the search query. Indeed, **the attributes present in `_formatted` is the addition of the attributes present in `attributesToRetrieve`, `attributesToHighlight`, and `attributesToCrop`**.
+
+Kmowing the default value of `attributesToRetrieve` is `["*"]` (so all the attributes present in `displayedAttributes`), if no `attributesToRetrieve` are set in the search query, `_formatted` will return all the `displayedAttributes`.
+
+Returning attributes in the `_formatted` object does not mean these attributes will be necessarily highlighted or cropped, see the next point.
 
-- If `attributesToHighlight` and `attributesToCrop` are not set, `_formatted` is not returned.
+Which attributes are highlighted or cropped in `_formatted`?
+
+No matter how many attributes are retrieved in `_formatted` following the previous rule:
+- Only the attributes present in `attributesToHighlight` are highlighted.
+- Only the attributes present in `attributesToCrop` are cropped.
+- Attributes present in both are cropped and highlighted at the same time.
+
+Some edge cases:
 - If cumulated fields in `attributesToHighlight` and `attributesToCrop` resolve to only having non-existent fields, `_formatted` is not returned.
-- If `attributesToRetrieve` is equal to `*` and `attributesToHighlight` or `attributesToCrop` are equals to `*`, `_formatted` is returned and contains `displayedAttributes` setting fields then compute highlights and crops on each received fields.
-- If `attributesToRetrieve` is equal to `*` and `attributesToHighlight` or `attributesToCrop` contains a set of fields, `_formatted` is returned and contains `displayedAttributes` setting fields but only compute highlights and crops on fields declared in `attributesToHighlight` or `attributesToCrop`.
-- If a list of fields is defined for `attributesToRetrieve` and `attributesToHighlight` / `attributesToCrop` are equals to `*`, `_formatted` is returned and contains `displayedAttributes` setting fields then compute highlights and crops on each received fields.
-- If a list of fields is defined for `attributesToRetrieve` and `attributesToHighlight` / `attributesToCrop` contains a list of fields, `_formatted` is returned and contains `attributesToRetrieve` fields, plus the fields set in `attributesToHighlight` or `attributesToCrop` then compute highlights and crops only for fields defined in `attributesToHighlight` / `attributesToCrop` parameters.
+
+Some examples:
```

Comment:
The examples are here to help, but do NOT replace the explanation: there are just here to illustrate the sentences above.
The sentences above should resume the previous list you did before.

Reply:
IMO It's much clearer that way, I think it's inevitable to have examples even if I like to avoid them to avoid making the specifications bigger. Thank you for this!